docs/codeguide/js.md

```diff
@@ -0,0 +1,140 @@
+# JavaScript
+
+## Important Concepts & Rules
```

Comment:
I think this section can be split into sub-sections with H3s (`###`).
Note that most can still under this `##` heading.
Up to you what you want to split them up to but some suggestions:
- `this` keyword
- Variable declaration